### PR TITLE
Update parameter queries with project filters

### PR DESCRIPTION
**Pull request by rickporter-tuono**
This is a draft, since it does not compile!

In addition to the query changes, I will need to update the upsert for adding parameters.

In most of the errors, I get complaints about `.parameters` is an unknown field in the query.

### CODE REVIEW

**Comment by rickporter-tuono** [comment]
Here's a sample of the output:

```
(new-host-4):~/cloudtruth-cli $ cargo run -q -- param ls -v
+----------------+---------------+---------+---------------------------------+
| Name           | Value         | Source  | Description                     |
+----------------+---------------+---------+---------------------------------+
| ADDED_FROM_CLI | set from cli  | default |                                 |
| ANOTHER_VALUE  | uninteresting | default |                                 |
| MY_KEY_NAME    | my_value      | default | dummy key name/value            |
| MY_SECRET      | *****         | default | something to test secret hiding |
+----------------+---------------+---------+---------------------------------+
(new-host-4):~/cloudtruth-cli $ cargo run -q -- --project Proj2 param ls -v
+---------------+------------------------+---------+-------------+
| Name          | Value                  | Source  | Description |
+---------------+------------------------+---------+-------------+
| MORE_FROM_CLI | set from cli with proj | default |             |
+---------------+------------------------+---------+-------------+
(new-host-4):~/cloudtruth-cli $ cargo run -q -- --project Proj2 param set CLI_ARG "more stuff"
Successfully updated parameter 'CLI_ARG' in environment 'default'.
(new-host-4):~/cloudtruth-cli $ cargo run -q -- --project Proj2 param ls -v
+---------------+------------------------+---------+-------------+
| Name          | Value                  | Source  | Description |
+---------------+------------------------+---------+-------------+
| CLI_ARG       | more stuff             | default |             |
| MORE_FROM_CLI | set from cli with proj | default |             |
+---------------+------------------------+---------+-------------+
(new-host-4):~/cloudtruth-cli $ 
```

File: src/graphql.rs [hunk 1]
```diff
@@ -65,6 +65,7 @@ pub enum GraphQLError {
     ServerError,
     UnauthorizedError(Resource, Operation),
     ValidationError(String, String),
+    ProjectNotFoundError(String),
```
Comment (nirvdrum) [inline src/graphql.rs:68]: I don't know if there's a way to enforce this in Clippy, but the enum variants are sorted.

File: src/graphql.rs [hunk 2]
```diff
@@ -135,7 +136,10 @@ impl fmt::Display for GraphQLError {
             ),
             GraphQLError::ServerError => write!(f, "There was an error on our server handling your request.\nOur ops team has been alerted and is investigating the issue."),
             GraphQLError::UnauthorizedError(resource, operation) => write!(f, "The access token is not authorized to {} this {}.", operation, resource),
-            GraphQLError::ValidationError(field, message) => write!(f, "There was a problem with a value you supplied: {} {}.", field, message)
+            GraphQLError::ValidationError(field, message) => write!(f, "There was a problem with a value you supplied: {} {}.", field, message),
+            GraphQLError::ProjectNotFoundError(name) => {
```
Comment (nirvdrum) [inline src/graphql.rs:140]: I don't know if we can enforce this with Clippy, but the match branches were sorted by enum variant.

File: src/graphql.rs [hunk 3]
```diff
@@ -135,7 +136,10 @@ impl fmt::Display for GraphQLError {
...
+                write!(f, "Unable to find project '{}'", name)
```
Comment (nirvdrum) [inline src/graphql.rs:141]: The other error messages use punctuation. We can drop that if you prefer, but the messages should be consistent with each other.

**Review by nirvdrum — APPROVED** [review]

File: src/graphql.rs (hunk 1 above)
Comment (rickporter-tuono) [reply src/graphql.rs]: I didn't see any support in clippy for catching this, but I did fix it.

File: src/graphql.rs (hunk 2 above)
Comment (rickporter-tuono) [reply src/graphql.rs]: I didn't find a way for Clippy to enforce it, but I fixed it.

File: src/graphql.rs (hunk 3 above)
Comment (rickporter-tuono) [reply src/graphql.rs]: Good catch.

**Pull request merged**